src/all/mangapluscreators/src/eu/kanade/tachiyomi/extension/all/mangapluscreators/MangaPlusCreators.kt

```diff
@@ -16,94 +18,200 @@ import okhttp3.Request
+            thumbnail_url = titleThumbnailUrl
+            setUrlWithoutDomain("/titles/$titleContentId")
+        }
+    }
+
+    // LATEST Section
     override fun latestUpdatesRequest(page: Int): Request {
         val newHeaders = headersBuilder()
             .set("Referer", "$baseUrl/titles/recent/?t=episode")
-            .add("X-Requested-With", "XMLHttpRequest")
             .build()
 
-        val apiUrl = "$API_URL/titles/recent/list".toHttpUrl().newBuilder()
+        val apiUrl = "$API_URL/titles/recent/".toHttpUrl().newBuilder()
             .addQueryParameter("page", page.toString())
-            .addQueryParameter("pageSize", POPULAR_PAGE_SIZE)
             .addQueryParameter("l", lang)
-            .addQueryParameter("c", "episode")
-            .addQueryParameter("isWebview", "false")
-            .addQueryParameter("_", System.currentTimeMillis().toString())
+            .addQueryParameter("t", "episode")
             .toString()
 
         return GET(apiUrl, newHeaders)
     }
 
-    override fun latestUpdatesParse(response: Response): MangasPage = popularMangaParse(response)
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val result = json.decodeFromString<MpcResponse>(response.body.string())
 
-    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val refererUrl = "$baseUrl/keywords".toHttpUrl().newBuilder()
-            .addQueryParameter("q", query)
-            .toString()
+        val titles = result.titles.orEmpty().map { title -> title.toSManga() }
 
-        val newHeaders = headersBuilder()
-            .set("Referer", refererUrl)
-            .add("X-Requested-With", "XMLHttpRequest")
-            .build()
+        // TODO: handle last page of latest
+        return MangasPage(titles, result.status != "error")
+    }
 
-        val apiUrl = "$API_URL/search/titles".toHttpUrl().newBuilder()
-            .addQueryParameter("keyword", query)
-            .addQueryParameter("page", page.toString())
-            .addQueryParameter("pageSize", POPULAR_PAGE_SIZE)
-            .addQueryParameter("sort", "newly")
+    fun MpcTitle.toSManga(): SManga {
+        val mTitle = this.title
+        val mAuthor = this.author.name // TODO: maybe not required
+        return SManga.create().apply {
+            title = mTitle
+            thumbnail_url = thumbnail
+            setUrlWithoutDomain("/titles/${latestEpisode.titleConnectId}")
+            author = mAuthor
+        }
+    }
+
+    // SEARCH Section
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        // TODO: HTTPSource::fetchSearchManga is deprecated? super.getSearchManga
+        if (query.startsWith(PREFIX_TITLE_ID_SEARCH)) {
+            val titleContentId = query.removePrefix(PREFIX_TITLE_ID_SEARCH)
+            val titleUrl = "$baseUrl/titles/$titleContentId"
+            return client.newCall(GET(titleUrl))
```

Comment:
pass `headers` to `GET` everywhere

Reply:
good point, will do. Wonder if you can avoid that in some other way 🤔